Scrapping_part_3/scrapping_part3/scrapping_part3/middlewares.py

```diff
@@ -0,0 +1,59 @@
+from scrapy import crawler
+
+from stem import Signal
+from stem.control import Controller
+from scrapy.downloadermiddlewares.httpproxy import HttpProxyMiddleware
+from stem.util.log import get_logger
+
+
+
+
+# Number of requests sent during the current tor identity
+
+
+
+def new_tor_identity():
+    with Controller.from_port(port=9051) as controller:
+        # authenticate with the tor password
+        file = open('password.txt')
```

Comment:
Bad practice to left a file descriptor open. Also it would be better to build absolute file path rather than relative.